lightweightsemaphore.h

```diff
@@ -0,0 +1,443 @@
+// Provides an efficient implementation of a semaphore (LightweightSemaphore)
+// - a synchronization primitive. This construct has been designed for 
+// threads and not processes.
+// ©2015-2016 Cameron Desrochers. Distributed under the terms of the simplified
```

Comment:
Since this is based on Jeff Preshing's code, I think his code's license should be moved to the top instead. I suppose you could keep a line metnioning that it has been "adapted and extended by Cameron Desrochers", but the important thing is that this is Jeff's code and all the changes I make to it are under the umbrella of his license, not mine.

The LICENSE file will need tweaking too, to reference the semaphore code by the new filename.